web/src/services/api.ts

```diff
@@ -371,4 +386,21 @@ export default class API {
       }`
     );
   }
+  // check whether or not is the first invite
+  fetchInviteStatus(): Promise<{ firstInvite: boolean; hasAchieved: boolean }> {
+    return this.fetch(
+      `${API_PATH}/challenge/${this.user.account.enrollment.challenge}/achievement/invite`
+    );
+  }
+
+  // Tell back-end user get unexpected achievement: invite + contribute in the same session
+  // Each user can only get once.
+  setInviteContributeAchievement(): Promise<void> {
+    return this.fetch(
+      `${API_PATH}/challenge/${this.user.account.enrollment.challenge}/achievement/session`,
+      {
+        method: 'POST',
+      }
+    );
+  }
```

Comment:
As above: I think we can achieve this in one round-trip if we send the relevant `sessionStorage` keys up with the initial request.